catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/data/QueryTemplateMetacard.java

```diff
@@ -110,6 +112,10 @@ public void setQuerySettings(Map<String, Object> json) {
     if (sources != null) {
       setAttribute(new AttributeImpl(QUERY_SOURCES, sources));
     }
+    List<Serializable> facets = safeGetList(json, FACETS, Serializable.class);
+    if (facets != null) {
```

Comment:
Maybe it doesn't matter, but should we also check that facets isn't an empty array as well?